configure.ac

```diff
@@ -440,6 +440,18 @@ else
   AM_CONDITIONAL(WITH_PCRE, false)
 fi
 
+dnl systemd structured logging
+
+AC_ARG_WITH([systemd-logging], [AS_HELP_STRING([--with-systemd-logging[[=PATH]]], [support systemd structured logging])], [], [with_systemd_logging=check])
+
+if test "x$with_systemd_logging" != xno
+then
+   CF3_WITH_LIBRARY(systemd_logging, [
+      AC_CHECK_LIB(systemd, sd_journal_sendv, [SYSTEMD_LDFLAGS="-lsystemd"], [if test "x$with_systemd_logging" != xcheck; then AC_MSG_ERROR(Cannot find systemd library); fi])
+      AC_CHECK_HEADERS(systemd/sd-journal.h, [], [if test "x$with_systemd_logging" != xcheck; then AC_MSG_ERROR(Cannot find systemd headers); fi])
```

Comment:
This cannot cause `./configure` to fail, the library is not available everywhere where we build things.

Reply:
I'm not entirely sure how the `CF3_WITH_LIBRARY` macro works. But for me–when configuring my macOS, which does not have the library–the error is only printed if I specify the `--with-systemd-logging` option to the configure script. I believe how it works when you use `xcheck`, is that it adds the library if it's there by default, but errors if you implicitly specify the `--with-systemd-logging` option. I might be entirely wrong though.

Reply:
Oh, I missed the built-in `if test "x$with_systemd_logging"` check! I'm used to doing the check after `AC_CHECK_HEADERS()`. Sorry for the noise.